src/NodeAnalyzer/CallLikeExpectsThisBindedClosureArgsAnalyzer.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\NodeAnalyzer;
+
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\CallLike;
+use PhpParser\Node\Expr\Closure;
+use PHPStan\Reflection\ExtendedParameterReflection;
+use Rector\NodeTypeResolver\PHPStan\ParametersAcceptorSelectorVariantsWrapper;
+use Rector\PHPStan\ScopeFetcher;
+use Rector\Reflection\ReflectionResolver;
+
+final readonly class CallLikeExpectsThisBindedClosureArgsAnalyzer
+{
+    public function __construct(
+        private ReflectionResolver $reflectionResolver
+    ) {
+    }
+
+    /**
+     * @return Arg[]
+     */
+    public function getArgsUsingThisBindedClosure(CallLike $callLike): array
+    {
+        $args = [];
+        $reflection = $this->reflectionResolver->resolveFunctionLikeReflectionFromCall($callLike);
+
+        if ($callLike->isFirstClassCallable()) {
+            return [];
+        }
+
+        if ($reflection === null) {
+            return [];
+        }
+
+        $scope = ScopeFetcher::fetch($callLike);
```

Comment:
using `$callLike->getAttribute(AttributeKey::SCOPE)` and check instanceof `\PHPStan\Analyser\Scope` seems fine here to ensure no issue on `tests/Issues`